src/modules/data.js

```diff
@@ -0,0 +1,29 @@
+const Data = (() => {
+  const todo = (title, desc = 'No description', date, completed, priority) => ({
+    title,
+    desc,
+    date,
+    completed,
+    priority,
+  });
+
+  const project = (title, todos = []) => ({
+    title,
+    todos,
+  });
+
+  const projectList = () => ({});
+
```

Comment:
### [observation]
i followed through your logic though, ideally, it will be great to have separate
`todo` and `project` factory functions,
instead of having one big massive Data structure like you do. This will work but, Anytime you have to affect a TASK, you have to search through. It's logically inefficient

Reply:
Hello Shey, 

Thank you for you time. Also I am sorry for the trouble, our instructions were not complete. 
Regarding your observation about our logic. I agree with you, the ideal way is to separate the projects from todos. But we didn't do it this way for 2 reasons. 
1. We are doing a MVC pattern here and more specifically the revealing module design pattern. Our controller receives two modules as arguments and works with them. This way when we want to change the looks of our app we will change the View module, we can name it whatever name and we just pass it in controller. It will work without any other change if we keep its API the same. Same can be done with Data module. 
2. Our data module is 29 lines long. So, I think, there is not much going on in it to justify creating another module. 

Considering the above reasons please let me know if you think it is better to separate project and todos functions.

Best Regards, 

Nick